### PR TITLE
fix: APP-234 create sell order

### PR DESCRIPTION
**Pull request by blushi**
## Description

https://regennetwork.atlassian.net/browse/APP-234

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

From https://deploy-preview-2410--regen-marketplace.netlify.app/
Log in with an account that has some ecocredits and go to your portfolio.
Create a new sell order leaving the checkbox "Require that credits are retired upon purchase" unchecked, then go to the storefront page, the purchase options of the sell order should be "tradable" and "retirable".
Create another sell order and check the checkbox, the only purchase option of the sell order should be "retirable".

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-marketplace/src/components/organisms/CreateSellOrderForm/CreateSellOrderForm.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ const CreateSellOrderForm: React.FC<Props> = ({
     price: undefined,
     askDenom: undefined,
     amount: undefined,
-    enableAutoRetire: true,
+    enableAutoRetire: false,
```
Comment (blushi) [inline web-marketplace/src/components/organisms/CreateSellOrderForm/CreateSellOrderForm.tsx:58]: This is the main fix.
The rest of the changes are about fixing the sell orders extended query to fetch *all* sell orders and not only the first 100 ones (that the ledger returns by default) so the fix can be tested with redwood on dev and deploy previews (redwood has more than 100 active sell orders so new sell orders weren't being displayed)

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -16,6 +16,8 @@ jobs:
       - name: install
         run: |
           bun install
+        with:
+          bun-version: 1.1.13
```
Comment (blushi) [inline .github/workflows/main.yml:20]: This is for github CI build step to work. This change is already on `dev`, there's some issues with bun latest versions so using 1.1.13 for now.

**Comment by blushi** [comment]
@erikalogie see testing instructions

**Comment by erikalogie** [comment]
@blushi I'm having trouble making the sell order, see this video:

Uploading Screen Recording 2024-07-23 at 5.26.38 PM.mov…

**Comment by blushi** [comment]
> @blushi I'm having trouble making the sell order, see this video:
> 
> Uploading Screen Recording 2024-07-23 at 5.26.38 PM.mov…

Looks like your video didn't get uploaded.
What if you reload the page and try again?

**Comment by erikalogie** [comment]
Ok it appears to be working well now, not sure what was happening before. Only thing to not is that I believe the box was checked before by default for "Require that credits are retired upon purchase". Can we bring that part back?

**Comment by erikalogie** [comment]
LGTM

**Review by wgardiner — APPROVED** [review]
Looks good. Tested in the netlify preview.

**Pull request merged**